langserver/references.go

```diff
@@ -105,6 +105,11 @@ func (h *LangHandler) handleTextDocumentReferences(ctx context.Context, conn jso
 		close(locsC)
 	}()
 
+	// Don't include decl if it is outside of workspace.
+	if params.Context.IncludeDeclaration && PathHasPrefix(defpkg, h.init.RootImportPath) {
```

Comment:
only this goroutine sends to the chan. Also the chan is not buffered, so this will block anyways until the collector collects it. I'll update to send this ref straight away.